pkg/apis/garden/validation/validation.go

```diff
@@ -1065,9 +1065,22 @@ func validateBackup(backup *garden.Backup, fldPath *field.Path) field.ErrorList
 	if backup == nil {
 		return allErrs
 	}
-	if _, err := cron.ParseStandard(backup.Schedule); err != nil {
+	if schedule, err := cron.ParseStandard(backup.Schedule); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("schedule"), backup.Schedule, "schedule must be in standard cron format"))
+	} else {
+		var (
+			nextScheduleTime              = schedule.Next(time.Now())
+			scheduleTimeAfterNextSchedule = schedule.Next(nextScheduleTime)
+			granularity                   = scheduleTimeAfterNextSchedule.Sub(nextScheduleTime)
+		)
+
+		if granularity < garden.MinimumETCDFullBackupTimeInterval {
```

Comment:
From API compatibility point of view hard-coding this (`garden.MinimumETCDFullBackupTimeInterval`) is a no-go. Better externalize it and make it configurable.

Reply:
Thing is, if we make it configurable then we are basically allowing backups to happen in an inefficient manner. These values are suggested after monitoring the side effects.

Reply:
> backups to happen in an inefficient manner.

This is up to the cluster admin to decide.

Reply:
@mvladev  What do you think, does it make sense to do that in  another PR? This PR mainly solves #550

Reply:
I am fine with removing this restriction completely. But things need to be documented there. We should warning comments over API related to low interval/high frequency for backups

Reply:
@swapnilgm how should we continue now? The problem is: The backups are managed in the accounts of the Gardener providers. As we allow configuration of backup schedule and maximum in shoot manifest we should somehow restrict the values so that users cannot blow the buckets by e.g. configuring too frequent or too many backups.
From my PoV we have the following options:
1. Deprecate `.spec.backup` and don't allow any modifications in the shoot manifest at all
1. Restrict `.spec.backup` so that only meaningful values can be entered (e.g., not too frequent backups) - as @mvladev suggested we could make the minimum etcd full backup time interval configurable in the controller-manager's componentconfig (although I don't like it too much as this value is depending on the etcd-backup-restore version, but I see the point regarding API compatibility)).

WDYT?

Reply:
ping @swapnilgm

Reply:
Was out-of-office because of D-kom. Anyways, let's deprecate it. both points blowing backup buckets and api incompatibility are quite strong arguments.

Reply:
OK, agreed.
@zanetworker could you please adapt the PR accordingly?
/cc @vlerenc

Reply:
Just to be clear on this: To deprecate on shoot means, we should move `shoot.spec.backup` to `gardener-controller-manager-component-configuration` .  Right @rfranzke?

Reply:
I don't think it should be configurable - meaningful values depend on the used etcd-backup-restore version, right?
We cannot immediately remove `.spec.backup` from `Shoot` resource (due to API compatibility reasons), but we should mark it as deprecated and not use the values anymore.

Reply:
> We cannot immediately remove .spec.backup from Shoot resource (due to API compatibility reasons), but we should mark it as deprecated and not use the values anymore.

Of course. I don't meant to remove it immediately. Agreed completely with this point.

> I don't think it should be configurable - meaningful values depend on the used etcd-backup-restore version, right?

IMO `spec.backup`  i.e backup schedule should still be configurable on `gardener-component-controller-manager configuration`. May be recommended value change per etcd-backup-restore version. But such parameters are more impacted by actual workload on the deployment. E.g. we initially set the schedule to run to frequent but then we updated it to low frequency looking at our workload behaviour of irrespective etcd-backup-restore version change. For schedule changes we won't want to make new release of gardener.

Reply:
OK, agreed - good arguments, thank you. @zanetworker please apply. :)

Reply:
Ok, closing this issue and will re-open a new one.